src/zm_monitor.cpp

```diff
@@ -2743,23 +2743,27 @@ void Monitor::TimestampImage(Image *ts_image, SystemTimePoint ts_time) const {
   while (*s_ptr && ((unsigned int)(d_ptr - label_text) < (unsigned int) sizeof(label_text))) {
     if ( *s_ptr == config.timestamp_code_char[0] ) {
       bool found_macro = false;
+      int rc = 0;
       switch ( *(s_ptr+1) ) {
         case 'N' :
-          d_ptr += snprintf(d_ptr, sizeof(label_text)-(d_ptr-label_text), "%s", name.c_str());
+          rc = snprintf(d_ptr, sizeof(label_text)-(d_ptr-label_text), "%s", name.c_str());
```

Comment:
Maybe pull out the `sizeof(label_text)-(d_ptr-label_text)` calculations. GCC with -O2 seems to generate shorter assembly and quick benchmarking shows a 10% improvement there.